proxy/src/serverless/sql_over_http.rs

```diff
@@ -405,6 +406,22 @@ impl UserFacingError for SqlOverHttpError {
     }
 }
 
+impl HttpCodeError for SqlOverHttpError {
+    fn get_http_status_code(&self) -> StatusCode {
+        match self {
+            SqlOverHttpError::ReadPayload(_) => StatusCode::BAD_REQUEST,
+            SqlOverHttpError::ConnectCompute(_) => StatusCode::INTERNAL_SERVER_ERROR,
+            SqlOverHttpError::ConnInfo(_) => StatusCode::INTERNAL_SERVER_ERROR,
+            SqlOverHttpError::RequestTooLarge(_) => StatusCode::PAYLOAD_TOO_LARGE,
+            SqlOverHttpError::ResponseTooLarge(_) => StatusCode::INTERNAL_SERVER_ERROR,
```

Comment:
I'll have to think about this one. 🤔

Reply:
maybe https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/507 ?

Reply:
... or https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/422
> indicates that the server understood the content type of the request content, and the syntax of the request content was correct, but it was unable to process the contained instructions.

> Clients that receive a 422 response should expect that repeating the request without modification will fail with the same error.

Reply:
I like 507